Lesson2/BankAccount.cs

```diff
@@ -0,0 +1,54 @@
+﻿//Для физических лиц
+public enum TypeBankAccount
+{
+    Current,
+    Accounting,
+    Credit,
+    Deposit,
+}
+
+public class BankAccount
+{
+
+    private static int accountNumber = 1;
+
+    private int _number;
+
+    public decimal Balance { get; set; }
+
+    public TypeBankAccount Type { get; set; }
+
+    public int Number
+    {
+        get => _number;
+        set
+        {
+            _number = value;
+        }
+    }
```

Comment:
```suggestion
    public decimal Balance { get; /* private */ set; }

    public TypeBankAccount Type { get; private set; }

    public int Number
    {
        get => _number;
        private set
        {
            _number = value;
        }
    }
```

Не очень хорошо когда "снаружи" класса можно запросто поменять тип счёта. Тип счёта должен фиксироваться на момент создания счёта. Если надо сменить тип счёта, то надо создать счёт нового типа и перевести на него деньги со старого. Номер счёта, тем более, менять нельзя после его создания!